src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -558,19 +558,32 @@ private void AttributeValue(HtmlSymbolType quote)
 
             if (At(HtmlSymbolType.Transition))
             {
-                var valueStart = CurrentLocation;
-                PutCurrentBack();
+                if (NextIs(HtmlSymbolType.Transition))
+                {
+                    // Render a single "@" in place of "@@".
+                    Output(SpanKind.Markup);
+                    AcceptAndMoveNext();
+                    Span.CodeGenerator = SpanCodeGenerator.Null;
+                    Output(SpanKind.Markup, AcceptedCharacters.None);
+                    AcceptAndMoveNext();
+                    Output(SpanKind.Markup, AcceptedCharacters.None);
```

Comment:
@dougbu, I tried looking at the places we special case transitions in [`CSharpCodeParser.Statements`](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs#L493), [`CSharpCodeParser`](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs#L162) and [`HtmlMarkupParser`](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs#L86) and I could not find any similar issue.